### PR TITLE
groups: use existence checks to ensure we can scry

### PR DESCRIPTION
**Pull request by arthyn**
We weren't checking whether or not a channel existed before scrying so this caused any updates that deleted roles to crash, causing a kick-resubscribe loop. This should fix the resubscribe loop for those who are receiving updates.  

Fixes LAND-675

### CODE REVIEW

File: desk/app/groups.hoon [hunk 1]
```diff
@@ -254,6 +254,8 @@
     ?.  (gth ~(wyt in readers) 0)  cr
     =/  action  [flag now.bowl %channel nest %del-sects readers]
     cr(cards [[%pass /groups/role %agent [our.bowl dap.bowl] %poke [act:mar:g !>(action)]] cards.cr])
+  =+  .^(has=? %gu (welp (channel-scry nest) ~))
```
Comment (Fang-) [inline desk/app/groups.hoon:257]:
```suggestion
  =+  .^(has=? %gu (channel-scry nest))
```

Also consider just inlining this `.^` into the `?.`.

File: desk/sur/groups.hoon [hunk 2]
```diff
@@ -3,7 +3,7 @@
 /-  grp=group-store
 /-  metadata-store
 |%
-++  okay  `epic:e`1
```
Comment (Fang-) [inline desk/sur/groups.hoon:6]: Not clear to me what's up with this and the v3 marks. We didn't change the types or subscription API, right?

File: desk/app/chat.hoon [hunk 3]
```diff
@@ -772,21 +772,23 @@
...
-      ``flag+!>(|)
+      ``loob+!>(|)
     (di-peek:(di-abed:di-core ship) %u t.t.t.path)
   ::
       [%u %club @ *]
     =/  =id:club:c  (slav %uv i.t.t.path)
     ?.  (~(has by clubs) id)
-      ``flag+!>(|)
+      ``loob+!>(|)
     (cu-peek:(cu-abed:cu-core id) %u t.t.t.path)
```
[the remark this answers is not in the document]
Comment (arthyn) [reply desk/app/chat.hoon:782]: might need to get on gather for this

**Review by philipcmonk — CHANGES_REQUESTED** [review]
This looks good, except I think we need to add `%group-action-2` in +poke and change the tags in +go-take-update.

How has this been tested?

**Comment by arthyn** [comment]
bah thought I pushed that change ty @philipcmonk

**Comment by arthyn** [comment]
@philipcmonk before I added the new marks, this was tested by having one moon with channels that are protected and not joinable along with a some extra roles existing. a second moon joins the group. the original moon then deletes a role which triggers `verify-group-cabals`.

**Comment by philipcmonk** [comment]
Great.  Have you tested it after adding the new marks?  At least to ensure that two moons with the changes can get groups updates from each other, and that they don't get updates from old ships (similar to our tests last time).

**Comment by arthyn** [comment]
not yet attempting now

File: desk/sur/groups.hoon (hunk 2 above)
Comment (arthyn) [reply desk/sur/groups.hoon:6]: yeah this was the reason, no state changes just a semi-convenient way to break the subscription

**Comment by arthyn** [comment]
@philipcmonk testing looks good

**Review by philipcmonk — APPROVED** [review]
Great, then this looks good

**Comment by arthyn** [comment]
I have a separate branch not on develop with the same changes that will be deployed: https://github.com/tloncorp/landscape-apps/tree/release-cabal-hotfix, but I'll merge this so this can be in the next release too

**Pull request merged**